cpp/vulkan_gui/CMakeLists.txt

```diff
@@ -0,0 +1,105 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+if(NOT IREE_TARGET_BACKEND_VULKAN_SPIRV OR
+   NOT IREE_HAL_DRIVER_VULKAN)
+  message(STATUS "Missing Vulkan backend and/or driver, skipping vulkan_gui sample")
+  return()
+endif()
+
+# This target statically links against Vulkan.
+# One way to achieve this is by installing the Vulkan SDK from
+# https://vulkan.lunarg.com/.
+include(FindVulkan)
+if(NOT Vulkan_FOUND)
+  message(STATUS "Could not find Vulkan, disabling Vulkan GUI programs")
+  return()
+endif()
+
+# vcpkg install imgui[vulkan-binding,sdl2-binding]
+#   tested with versions 1.83 - 1.87#1
+find_package(imgui CONFIG QUIET)
+if(NOT imgui_FOUND)
+  message(STATUS "Could not find Dear ImGui, disabling Vulkan GUI programs")
+  return()
+endif()
+
+# vcpkg install sdl2[vulkan]
+#   tested with versions 2.0.14#4 - 2.0.22#1
+find_package(SDL2 CONFIG QUIET)
+if(NOT SDL2_FOUND)
+  message(STATUS "Could not find SDL2, disabling Vulkan GUI programs")
+  return()
+endif()
+
+# Compile simple_mul.mlir to simple_mul.vmfb.
```

Comment:
Note that the namespace changes in `iree_macros.cmake` broke using `iree_bytecode_module`, `iree_cc_library`, etc. out of tree (at least at the path that `FetchContent` is using). I'm _sort of_ okay with that, since those are really for internal project use, but I could also see an argument for making those work again with some way to set an explicit namespace.